hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java

```diff
@@ -97,19 +99,60 @@ public static boolean isNotEmpty(Collection collection) {
     }
 
     /**
-     * @param collection           collection of objects to be converted to collection of data
-     * @param serializationService will be used for converting object to data
+     * Converts a collection of any type to a collection of {@link Data}.
+     *
+     * @param collection           the given collection
+     * @param serializationService will be used for converting object to {@link Data}
      * @return collection of data
-     * @throws java.lang.NullPointerException if collection is null, or contains a null element
+     * @throws java.lang.NullPointerException if collection is {@code null}, or contains a {@code null} item
      */
-    public static <C> Collection<Data> objectToDataCollection(Collection<C> collection
-            , SerializationService serializationService) {
+    public static <C> Collection<Data> objectToDataCollection(Collection<C> collection,
+                                                              SerializationService serializationService) {
         List<Data> dataKeys = new ArrayList<Data>(collection.size());
-        for (C c : collection) {
-            Preconditions.checkNotNull(c);
-            dataKeys.add(serializationService.toData(c));
+        for (C item : collection) {
+            checkNotNull(item);
+            dataKeys.add(serializationService.toData(item));
         }
         return dataKeys;
     }
 
+    /**
+     * Converts a {@link Collection<Integer>} to a primitive {@code int[]} array.
+     *
+     * Returns an empty array if the collection is {@code null}.
+     *
+     * @param collection the given collection
+     * @return a primitive int[] array
+     */
+    public static int[] toIntArray(Collection<Integer> collection) {
+        if (collection == null) {
```

Comment:
Because otherwise we'll get a NPE here?
`int[] collectionArray = new int[collection.size()];`